File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    let hm=new Map()
    let len=nums.length;
    for(let i=0;i<len;i++){
        let x=nums[i]
        if(hm.has(x)){
            let abs=hm.get(x)-i
            if(abs<0) abs=0-abs
            if(abs<=k) return true;
        }
        hm.set(x,i)
    }
    return false;
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map <int,int> hm{};
        int length=nums.size();
        for(int i=0;i<length;i++){
            if(hm.count(nums[i])){
                if(abs(hm[nums[i]]-i)<=k){
                    return true;
                }
            }
            hm[nums[i]]=i;
        }
        return false;
    }
};